lib/checks/base_paths_missing_from_publishing_api.rb

```diff
@@ -0,0 +1,31 @@
+module Checks
+  class BasePathsMissingFromPublishingApi
+
+    def initialize(checker_db, whitelist)
+      @checker_db = checker_db
+      @whitelist = whitelist
+    end
+
+    # find base_paths present in Rummager which don't map to a published content item in the Publishing API
+
+    def run_check
+      query = <<-SQL
+      SELECT
+      rc.base_path,
+      rc.format,
+      rc.rummager_index,
+      rc.document_type
+      FROM rummager_content rc
+      LEFT JOIN rummager_base_path_content_id lookup ON rc.base_path = lookup.base_path
+      WHERE lookup.content_id IS NULL
+      AND format NOT IN ('recommended-link')
+      SQL
+
+      name = self.class.name.split('::').last
```

Comment:
I didn't bother before because the value is needed for the whitelist also. Really it's a property of the check, which should be extracted by the common check creation code. I'll address it there.